Packs/Arduino/Integrations/Arduino/Arduino.py

```diff
@@ -0,0 +1,162 @@
+                    return 'ok'
+                else:
+                    return f"There was an issue. Check the Arduino code - {data}"
+        except Exception as err:
+            return err
+
+    def send_data(self, value) -> str:
+        try:
+            with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+                s.connect((self.host, self.port))
+                s.sendall(value.encode())
+                data = s.recv(1024)
+                return data
+        except Exception as err:
+            return_error(err)
+
+
+''' HELPER FUNCTIONS '''
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def test_module(server: Server) -> str:
+    return server.test_connect()
+
+
+def arduino_set_pin_command(server: Server, args: any) -> str:
+    pin_type: str = args.get('pin_type')
+    prefix: str = "Arduino.DigitalPins" if pin_type == "digital" else "Arduino.AnalogPins"
+    pin_number: int = int(args.get('pin_number'))
+    value: int = int(args.get('value'))
+    result: int = int(server.send_data(f"set:{pin_type}:{pin_number},{value}"))
+    results = [{
+        "PinType": "Digital" if pin_type == "digital" else "Analog",
+        "PinNumber": pin_number,
+        "PinValue": result
+    }]
+    command_results = CommandResults(
+        outputs_prefix=prefix,
+        outputs_key_field=['PinNumber', 'PinType'],
+        outputs=results,
+        readable_output=tableToMarkdown(f"Set pin {pin_number} on {server.host}({server.port}):", results)
+    )
+    return command_results
+
+
+def arduino_get_pin_command(server: Server, args: any) -> str:
+    pin_type: str = args.get('pin_type')
+    prefix = "Arduino.DigitalPins" if pin_type == "digital" else "Arduino.AnalogPins"
+    pin_number: int = int(args.get('pin_number'))
+    result: int = int(server.send_data(f"get:{pin_type}:{pin_number}"))
+    results = [{
+        "PinType": "Digital" if pin_type == "digital" else "Analog",
+        "PinNumber": pin_number,
+        "PinValue": result
+    }]
+    command_results = CommandResults(
+        outputs_prefix=prefix,
+        outputs_key_field=['PinNumber', 'PinType'],
+        outputs=results,
+        readable_output=tableToMarkdown(f"Get pin {pin_number} on {server.host}({server.port}):", results)
+    )
+    return command_results
+
+
+def arduino_send_data_command(server: Server, args: any) -> str:
+    data: str = args.get('data')
+    result: str = server.send_data(data)
+    results = [{
+        "Sent": data,
+        "Received": result.decode()
+    }]
+    command_results = CommandResults(
+        outputs_prefix="Arduino.DataSend",
+        outputs_key_field='Sent',
+        outputs=results,
+        readable_output=tableToMarkdown(f"Data sent to {server.host}({server.port}):", results)
+    )
+    return command_results
+
+
+''' MAIN FUNCTION '''
+
+
+def main() -> None:
+
+    params: any = demisto.params()
+    host: str = params.get('host')
+    port: int = int(params.get('port'))
```

Comment:
Please pay attention that you would get a `ValueError` exception when a user inserts a string (for example 11a). You don't catch such exception in your function so the user will not know what caused the command to fail. I suggest to catch this kind of exception and return an informative error to the user.
In addition:
```suggestion
    port = int(params.get('port'))
```

Reply:
Have put some basic checking in place